tests/generic-java-signatures/17069.scala

```diff
@@ -0,0 +1,8 @@
+
+class Foo:
+  val generic: List[String] = ???
+
+@main def Test =
+  val tpe = classOf[Foo].getDeclaredField("generic").getGenericType()
+  val tpe2 = classOf[Foo].getDeclaredMethod("generic").getGenericReturnType()
+  assert(tpe == tpe2)
```

Comment:
This test will still pass if *both* the field and getter have broken signatures. We should also actually test the structure of `tpe`.